assets/src/hooks/use_api_response.tsx

```diff
@@ -2,24 +2,39 @@ import { useEffect, useState } from "react";
 import { isDup } from "Util/util";
 import useInterval from "Hooks/use_interval";
 import { useLocation } from "react-router-dom";
+import * as Sentry from "@sentry/react";
 
 const MINUTE_IN_MS = 60_000;
 
-const FAILURE_RESPONSE = { success: false };
+const FAILURE_RESPONSE = { success: false, lastSuccess: null };
 
 const doFailureBuffer = (
-  lastSuccess: number,
+  lastSuccess: number | null,
   failureModeElapsedMs: number,
   setApiResponse: React.Dispatch<React.SetStateAction<object>>,
+  setLastResponseWasFailure: React.Dispatch<React.SetStateAction<boolean>>,
   apiResponse: object = FAILURE_RESPONSE
 ) => {
-  const elapsedMs = Date.now() - lastSuccess;
-
-  if (elapsedMs < failureModeElapsedMs) {
+  if (lastSuccess == null) {
+    // We haven't had a successful request since initial page load.
+    // Continue showing the initial "no data" state.
     setApiResponse((state) => state);
-  }
-  if (elapsedMs >= failureModeElapsedMs) {
-    setApiResponse(apiResponse);
+  } else {
+    const elapsedMs = Date.now() - lastSuccess;
+
+    // Because we recycle state when a failure occurs,
+    // we need to track failures in a separate state variable.
+    setLastResponseWasFailure(true);
+
+    if (elapsedMs < failureModeElapsedMs) {
+      setApiResponse((state) => state);
+    }
+    if (elapsedMs >= failureModeElapsedMs) {
+      setApiResponse(apiResponse);
```

Comment:
My thinking here was that the `captureMessage` on line 37 would tell us as soon as a failure occurs, regardless if the actual `NoData` page is being displayed. So even when stale data is being displayed, we know that a failure is happening. Would an additional log still be useful given that? I think it might be redundant.

Reply:
Based on the asana task title/description, it looks like we want to just log once when we transition into the state, and once when we transition out of it. I guess the idea is to get a better picture of when and for how long riders are actually seeing the no-data state.

Reply:
Gotcha. Makes total sense. I was just trying to do a little too much, but a log every failure is excessive. Got that changed.